scripts/k8s/setup-io-prereq.sh

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
+
+set -e
+
+HUGE_PAGES=
+HUGE_PAGES_OVERRIDE=
+NVME_TCP=
+DRY_RUN=
+SYSCTL="sudo sysctl"
+MODPROBE="sudo modprobe"
+help() {
+  cat <<EOF
+Usage: $(basename "$0") [COMMAND] [OPTIONS]
+
+Options:
+  -h, --help                            Display this text.
+  --hugepages         <num>             Add <num> 2MiB hugepages.
+  --nvme-tcp                            Load nvme_tcp kernel modules.
+
+Examples:
+  $(basename "$0") --nvme-tcp --hugepages 2048
+EOF
+}
+
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die() {
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
+
+setup_hugepages() {
+  $SYSCTL -w vm.nr_hugepages="$1"
+}
+
+modprobe_nvme_tcp() {
+  $MODPROBE nvme_tcp
+}
+
+nvme_ana_check() {
+  cat /sys/module/nvme_core/parameters/multipath
+}
+
+distro() {
+  cat /etc/os-release | awk -F= '/^NAME=/ {print $2}' | tr -d '"'
+}
+
+install_kernel_modules_nsup() {
+  die "Installing kernel modules not supported for $1"
+}
+
+install_kernel_modules() {
+  DISTRO="$(distro)"
+  case "$DISTRO" in
+    Ubuntu)
+      sudo apt-get install linux-modules-extra-$(uname -r)
+      ;;
+    NixOS | *)
+      install_kernel_modules_nsup "$DISTRO"
+      ;;
+  esac
+}
+
+install_zfs_and_lvm() {
```

Comment:
I think we should check if the tools are already present? In which case we can skip install?

Reply:
Also perhaps worth splitting zfs and lvm into 2